src/test/scala/com/crealytics/spark/excel/IntegrationSuite.scala

```diff
@@ -111,4 +114,21 @@ class IntegrationSuite extends FunSuite with PropertyChecks with DataFrameSuiteB
       assertDataFrameEquals(expectedWithEmptyStr, writeThenRead(expectedWithNull))
     }
   }
+
+  test("infers schema correctly") {
+    forAll(rowsGen, MinSuccessful(20)) { rows =>
+      val df = spark.createDataset(rows).toDF
+      val inferred = writeThenRead(df, schema = None)
+
+      assert(if (df.count() > 0) BooleanType else StringType, inferred.schema.fields(0).dataType)
```

Comment:
You could rewrite this to something like (untested):
```scala
val expectedDataTypes =
  if (df.count() == 0) {
    // Without actual data, we assume everything is a StringType
    exampleDataSchema.fields.map(_ => StringType)
  } else {
    exampleDataSchema.fields.map(_.dataType)
  }
assert(inferred.schema.fields.map(_.dataType), expectedDataTypes)

```

Reply:
OK I will tidy it up

Reply:
We cannot use the types from exampleDataSchema because the inferred types will not match exactly. For example, the byte, short, int and long columns will all be inferred as DoubleType.

Reply:
Ok, then maybe just make it a `Seq(DoubleType, DoubleType, ..., TimestampType)`

Reply:
Yep